addons/web/static/tests/views/fields/statusbar_time_field_tests.js

```diff
@@ -0,0 +1,112 @@
+/** @odoo-module **/
+
+import {click, getFixture} from "@web/../tests/helpers/utils";
+import { makeView, setupViewRegistries } from "@web/../tests/views/helpers";
+
+
+let serverData;
+let target;
+
+QUnit.module("Fields", (hooks) => {
+    hooks.beforeEach(() => {
+        target = getFixture();
+        serverData = {
+            models: {
+                partner: {
+                    fields: {
+                        display_name: { string: "Displayed name", type: "char" },
+                        stage_id: { string: "Stage", type: "many2one", relation: "stage_model" },
+                        json_statustime: { string: "Time per stage", type: "char", default: "{}"},
+                    },
+                    records: [
+                        {
+                            id: 1,
+                            display_name: "first record",
+                            stage_id: 30,
+                            json_statustime: {"10": 604800, "20": 10800},  // 7 days - 3 hours
+                        },
+                    ],
+                },
+                stage_model: {
+                    fields: {
+                        name: { string: "Stage Name", type: "char" },
+                    },
+                    records: [
+                        {
+                            id: 10,
+                            display_name: "New",
+                        },
+                        {
+                            id: 20,
+                            display_name: "Qualified",
+                        },
+                        {
+                            id: 30,
+                            display_name: "Proposition",
+                        },
+                        {
+                            id: 40,
+                            display_name: "Won",
+                        },
+
+                    ],
+                },
+            },
+        };
+
+        setupViewRegistries();
+    });
+
+    QUnit.module("StatusBarTimeField");
+
+    QUnit.test("statusbar time per stage tests", async function (assert) {
+        await makeView({
+            serverData,
+            type: "form",
+            resModel: "partner",
+            arch: `
+                    <form>
+                        <header>
+                            <field name="json_statustime" invisible="1" />
+                            <field name="stage_id" widget="statusbar_time" />
+                        </header>
+                    </form>`,
+            resId: 1,
+        });
+
+        // 1. check that time button is toggled off and exists
+        let timeButton = target.querySelector("div.o_statusbar_status div div.time-btn-container div.time-btn-hover");
+        assert.notEqual(timeButton, null);
```

Comment:
```suggestion
        assert.containsN(target, ".o_statusbar_status .btn-time", 4);
```
don't know if it's 4, but you can check the exact number of occurences that way. Also, no need to write complex selectors encoding the whole DOM, because it becomes a pain later when the DOM is reworked.